src/dialogs/mkdir.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react';
+import {
+    Button,
+    Form, FormGroup,
+    FormSelect, FormSelectOption,
+    Modal, ModalVariant,
+    Stack, TextInput,
+} from '@patternfly/react-core';
+
+import cockpit from 'cockpit';
+import { useDialogs } from 'dialogs';
+import { InlineNotification } from 'cockpit-components-inline-notification';
+import { FormHelper } from 'cockpit-components-form-helper';
+import { superuser } from 'superuser';
+
+import { useId, get_owner_candidates } from '../ownership';
+import { useFilesContext } from '../app';
+
+const _ = cockpit.gettext;
+
+function check_name(candidate: string) {
+    if (candidate === "") {
+        return _("Directory name cannot be empty.");
+    } else if (candidate.length >= 256) {
+        return _("Directory name too long.");
+    } else if (candidate.includes("/")) {
+        return _("Directory name cannot include a /.");
+    } else {
+        return undefined;
+    }
+}
+
+async function create_directory(path: string, owner?: string) {
+    console.log('mkdir', path, owner);
+    if (owner !== undefined) {
+        const opts = { err: "message", superuser: "require" } as const;
+        await cockpit.spawn(["mkdir", path], opts);
+        await cockpit.spawn(["chown", owner, path], opts);
+    } else {
+        await cockpit.spawn(["mkdir", path], { err: "message" });
+    }
+}
+
+export const CreateDirectoryModal = ({ currentPath }: { currentPath: string }) => {
+    const Dialogs = useDialogs();
+    const [name, setName] = useState("");
+    const [nameError, setNameError] = useState<string | undefined>();
+    const [errorMessage, setErrorMessage] = useState<string | undefined>();
+    const [owner, setOwner] = useState<string | undefined>();
+    const createDirectory = () => {
+        const path = currentPath + name;
+        create_directory(path, owner).then(Dialogs.close, err => setErrorMessage(err.message));
+    };
+    const id = useId();
+    const { cwdInfo } = useFilesContext();
+
+    const candidates = [];
+    if (superuser.allowed && id && cwdInfo) {
+        candidates.push(...get_owner_candidates(id, cwdInfo));
+        if (owner === undefined) {
+            setOwner(candidates[0]);
+        }
+    }
+
+    return (
+        <Modal
+          position="top"
+          title={_("Create directory")}
+          isOpen
+          onClose={Dialogs.close}
+          variant={ModalVariant.small}
+          footer={
+              <>
+                  <Button
+                    variant="primary"
+                    onClick={createDirectory}
+                    isDisabled={errorMessage !== undefined || nameError !== undefined}
```

Comment:
This needs to be disabled when `superuser.allowed && !id && !cwdInfo`

Reply:
```suggests
-                    isDisabled={errorMessage !== undefined || nameError !== undefined}
+                    isDisabled={errorMessage !== undefined || nameError !== undefined || (superuser.allowed && (!id && !cwdInfo))}
```

Reply:
I think you meant `(!id || !cwdInfo)` or `!(id && cwdInfo)`, yes?